torch_xla/torch_xla.py

```diff
@@ -83,9 +85,17 @@ def step():
   return compile()
 
 
-def compile(f: Optional[Callable] = None,
-            full_graph: Optional[bool] = False,
-            name: Optional[str] = None):
+# Keeps track of the alive functions. This allow us to remove session entries in the
+# C++ side for functions that are no longer alive.
+_compiled_id_to_functions_ref = weakref.WeakValueDictionary()
+
+
+def compile(
+    f: Optional[Callable] = None,
+    full_graph: Optional[bool] = False,
+    name: Optional[str] = None,
+    allowed_traces: Optional[int] = None,
```

Comment:
I think `traces` is too much of an implementation detail. Since we already have the `full_graph` above, how about `num_different_graph_allowed`?

Reply:
Wouldn't that be kind of confusing, leading the user to think of the number of HLO graphs?

Reply:
Ok. Couldn't think of something better, so went with your suggestion.

Reply:
Hmm. That's not the case if we have a fallback operation in the middle, is it?

Reply:
oh I see, I generally expect user to also turn on `full_graph` to True.. I guess `trace` is more correct but I just find it expose too much underlying implementation details.